Foundation/NSNumberFormatter.swift

```diff
@@ -7,18 +7,34 @@
 // See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 
-
+import CoreFoundation
 
 public class NSNumberFormatter : NSFormatter {
     
-    public override init() { NSUnimplemented() }
-    public required init?(coder aDecoder: NSCoder) {
-        NSUnimplemented()
+    typealias CFType = CFNumberFormatterRef
+    internal var __cfObject: CFType?
+    internal var _cfObject: CFType {
+        get {
+            if let obj = __cfObject {
+                return obj
+            } else {
+                #if os(OSX) || os(iOS)
+                    let numberStyle = CFNumberFormatterStyle(rawValue: CFIndex(self.numberStyle.rawValue))!
+                #else
+                    let numberStyle = CFNumberFormatterStyle(self.numberStyle.rawValue)
+                #endif
+                
+                let obj = CFNumberFormatterCreate(kCFAllocatorSystemDefault, locale._cfObject, numberStyle)
+                _setFormatterAttributes(obj)
+                __cfObject = obj
+                return obj
+            }
+        }
```

Comment:
👍 ef329ed

Regarding (2), what do you think of making it `private`?  If we need access to the internal formatter somewhere else in the lib, one still could use the `_cfFormatter`.